src/platform/site-wide/tests/component-library-analytics-setup.unit.spec.js

```diff
@@ -0,0 +1,29 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { subscribeComponentAnalyticsEvents } from 'platform/site-wide/component-library-analytics-setup';
+
+const recordEvent = sinon.spy();
+const event = {
+  detail: {
+    componentName: 'Modal',
+    action: 'show',
+    details: {
+      title: 'Modal title',
+      status: 'info',
+      primaryButtonText: 'Button Text 1',
+      secondaryButtonText: 'Button Text 3',
+    },
+  },
+};
+
+const dataLayerEvent = {
+  event: 'int-modal-click',
+  'modal-title': 'Modal title',
+  'modal-status': 'info',
+  'modal-primaryButtonText': 'Button Text 1',
+  'modal-secondaryButtonText': 'Button Text 3',
+};
+
+subscribeComponentAnalyticsEvents(event, recordEvent);
+
+expect(recordEvent.calledWith(dataLayerEvent)).to.be.true;
```

Comment:
Oh sorry, my code was untested and incomplete. I was just trying to show a PoC for getting started but it is up to you as far as what gives you confidence. We'll at least need to wrap the code in `describe` and `it` Mocha methods

Reply:
No worries - I changed it to `calledWith` and it's very close to the type of testing we're doing on the component-library side of things.

Updated with _describe_ / _it_.